game/world/objects/npc.cpp

```diff
@@ -1356,6 +1356,16 @@ bool Npc::implGoTo(uint64_t dt, float destDist) {
     if(go2.flag==GT_Way) {
       go2.wp = wayPath.pop();
       if(go2.wp!=nullptr) {
+        if(std::abs(go2.wp->x-x)+std::abs(go2.wp->z-z)<std::abs(go2.wp->y-y)) {
+          auto inter = owner.availableMob(*this,"LADDER",true);
+          if(inter!=nullptr) {
+            auto wpoint = wayPath.last()!=nullptr ? wayPath.last() : go2.wp;
+            clearGoTo();
+            aiQueue.pushFront(AiQueue::aiGoToPoint(*wpoint));
+            aiQueue.pushFront(AiQueue::aiUseMob("LADDER",0));
```

Comment:
Is there a simple way to implement same without poking into queue? Ai-Queue is for script side, and I'm not sure if poking onto it can cause observable issues with mods